drivers/w1/w1_max32.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2023-2024 Analog Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/w1.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control/adi_max32_clock_control.h>
+
+#include <wrap_max32_owm.h>
+
+#define DT_DRV_COMPAT adi_max32_w1
+
+LOG_MODULE_REGISTER(w1_max32, CONFIG_W1_LOG_LEVEL);
+
+struct max32_w1_config {
+	struct w1_master_config w1_config;
+	mxc_owm_regs_t *regs;
+	const struct pinctrl_dev_config *pctrl;
+	const struct device *clock;
+	struct max32_perclk perclk;
+	int internal_pullup;
+	int external_pullup;
+	int long_line_mode;
+};
+
+struct max32_w1_data {
+	struct w1_master_data w1_data;
+	uint8_t reg_device_config;
+};
+
+static int api_reset_bus(const struct device *dev)
+{
+	int ret;
+	const struct max32_w1_config *const cfg = dev->config;
+	mxc_owm_regs_t *regs = cfg->regs;
+
+	/* 0 if no 1-wire devices responded during the presence pulse, 1 otherwise */
+	ret = MXC_OWM_Reset();
+	if (ret > 0) {
+		/* Check OW pin input state due to presence detect pin seems not work well */
+		if (regs->ctrl_stat & MXC_F_OWM_CTRL_STAT_OW_INPUT) {
+			ret = 1; /* At least 1 device on the line */
+		} else {
+			ret = 0; /* no device on the line */
+		}
```

Comment:
Having a short look at the datasheet, and the hal. It seems `MXC_OWM_Reset()` waits until the reset is complete and returns the result of the stat presence detect flag. Which sounds reasonable. Is there a known issue/errata on a few socs or do your tests show that relying on the presence detect flag directly is not sufficient?

Reply:
I see thanks. Was it something with the zephyr w1_api tests or something you observed with the temperature driver?
I am asking as the w1 api test does not test very much, it tests a bit more than that everything compiles.

Reply:
Added these logic to it pass w1_api tests. Otherwise it was failed while executing some tests that execute reset.
![image](https://github.com/user-attachments/assets/02005f35-02c3-4c89-ada2-2b470dbb25d0)

Reply:
I have bought a board and finally got it delivered so I will have a closer look at this some time this week.